Trivial Mobile/movement optimizations

reaperrr opened the pull request:
I skimmed through the trait and related activities again to check for any overlooked low-hanging fruits, but didn't find anything noteworthy.

Still, there was some risk-less 'at worst won't make a difference' small fry stuff, so I'm filing it anyway.

reaperrr committed:
Cache unchanging values for MoveWithinRange

In theory, CandidateMovementCells could be called
every tick, so let's avoid creating the vars
every time.

reaperrr committed:
Add comment in Mobile

This isn't obvious to people not entirely familiar
with the code.

reaperrr committed:
Minor MoveOrderTargeter optimization

IsTraitPaused should be cheaper than Map.Contains,
so let's perform the cheaper check first.

reaperrr commented on OpenRA.Mods.Common/Traits/Mobile.cs:
```diff
@@ -804,9 +804,8 @@ Activity LocalMove(Actor self, WPos fromPos, WPos toPos, CPos cell)
-			if (CanEnterCell(above))
-				return above;
+			if (CanEnterCell(TopLeft))
+				return TopLeft;
```
I didn't bother profiling it, so I'm not sure if keeping the `var above` but just populating it with `TopLeft` directly might be even better.

pchote replied:
I expect this will break things: the reason for the `new CPos(TopLeft.X, TopLeft.Y)` was to reset any custom `TopLeft.Layer` back to the default ground cell!

reaperrr commented:
Updated, replaced 2nd commit with a comment explaining the purpose of that (so I won't make the same misunderstanding again in 2 years when I've forgotten all about it).

abcdefg30 reviewed (approved):
Ok

teinarss commented:
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/6d9e94d666f2bfd2fd63a7d64cc5409ce53438f2...81e2541984167fa3bbe7b29fa0d17e1d1d5c0cbb
